src/NuGet.Clients/NuGet.SolutionRestoreManager/VsSolutionRestoreService.cs

```diff
@@ -190,22 +205,39 @@ private static void DumpProjectRestoreInfo(PackageSpec packageSpec)
                 var dgPath = Path.Combine(outputPath, $"{Guid.NewGuid()}.dg");
                 dgFile.Save(dgPath);
             }
-            catch(Exception ex)
+            catch (Exception e)
             {
-                ExceptionHelper.WriteToActivityLog(ex);
+                _logger.LogError(e.ToString());
             }
         }
 
         private static PackageSpec ToPackageSpec(ProjectNames projectNames, IVsProjectRestoreInfo projectRestoreInfo)
         {
-            var tfis = projectRestoreInfo.TargetFrameworks
+            var tfis = projectRestoreInfo
+                .TargetFrameworks
                 .Cast<IVsTargetFrameworkInfo>()
                 .Select(ToTargetFrameworkInformation)
                 .ToArray();
 
             var projectFullPath = Path.GetFullPath(projectNames.FullName);
             var projectDirectory = Path.GetDirectoryName(projectFullPath);
 
+            var originalTargetFrameworks = tfis
+                .Select(tfi => tfi.FrameworkName.GetShortFolderName())
+                .ToList();
+            var crossTargeting = originalTargetFrameworks.Count > 1;
```

Comment:
In the `if` clause below the raw value of `OriginalTargetFrameworks` is examined. The initial "default" settings here were needed to avoid integration issues. Also, should be enable cross-targeting when multiple TFIs provided with no `$(TargetFrameworks)`?